src/arrayMethodJoin.js

```diff
@@ -3,10 +3,21 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(arg = ',') {
+    let parsedString = '';
```

Comment:
you are not really parsing anything here. 'res' or 'result' would be a better name.

Reply:
rewritten from scratch, not the best  even not good one :/